### PR TITLE
Template Selector Page and Layout Previews

### PR DESCRIPTION
**Pull request by melvinalex7**
# Description

A summary of the change
- Layout and Theme Selection moved to a new page, which can be accessed from the Researchify Dashboard's Sidebar.
- Each Layout Thumbnail On Hover displays a button for Layout Previews.
- Layout Previews Created for both Light and Dark Themes.

# Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation
- [x] Improvement

# Solution description

Describe your code changes in detail for reviewers.
Explain the technical solution you have provided and how it fixes the issue.

# Tests

All unit tests pass?
Y
Builds successfully?
Y
Do all SA checks pass?

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

- Select the Template Selector Option from the Sidebar on the Researchify Dashboard. 
- Hover over the Layout Thumbnails and click on Preview Button to open the Layout Previews.

# Relevant document/ link (if any)

Any related documents could help reviewers to review this PR

# Risk

Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low

### CODE REVIEW

**Comment by sanathmahesh** [comment]
> > Looks Great and works, but I cannot see the description for any layout, Is it a bug just for me? Screenshot (59)
> 
> Can you ensure that the branch is updated? This looks to be an intermediate commit during the development phase.

Yeah looks good now, my bad, I forgot to update the branch

**Review by jfan0010 — COMMENTED** [review]
some personal opinions: 
When first looking into the sidebar, I would think that every sidebar button is a page in my deployed GitHub page (except for the first dashboard button). If we put template selector in sidebar, I personally think that it would be great if we can somehow show the difference between the pages in client's GitHub deployed site and Researchify site. 
One example could be like (just an example):
image
Not sure if the drawing make sense and sorry for the terrible drawing 🤣

**Comment by melvinalex7** [comment]
> some personal opinions: When first looking into the sidebar, I would think that every sidebar button is a page in my deployed GitHub page (except for the first dashboard button). If we put template selector in sidebar, I personally think that it would be great if we can somehow show the difference between the pages in client's GitHub deployed site and Researchify site. One example could be like (just an example): image Not sure if the drawing make sense and sorry for the terrible drawing 🤣

Yea, even I have thought about it but it's something for a different PR and it's out of this PR's scope.

File: client/src/components/layouts/DashboardSidebarData.js [hunk 1]
```diff
@@ -45,6 +46,11 @@ const DashboardSidebarData = () => {
       link: '/achievement',
       name: 'ACHIEVEMENTS',
     },
+    {
+      title: 'Template Selector',
+      icon: <BsColumns />,
+      link: '/templates',
```
Comment (pmyip3) [inline client/src/components/layouts/DashboardSidebarData.js:52]: I think it should better placed under the Dashbroad ?

**Review by pmyip3 — COMMENTED** [review]
I remember iti right, the client said when the toggle is changed dark theme, the preview layout pictures should also be changed to dark theme too. Could you implement that as well

File: client/src/components/templateSelectorPage/LayoutThumbnail.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+              Preview
+            </Button>
+            <LayoutPreviewModal />
+          </div>
+        </Card.ImgOverlay>
+      </Card>
+    </>
+  );
+
+  function LayoutPreviewModal() {
```
Comment (pmyip3) [inline client/src/components/templateSelectorPage/LayoutThumbnail.js:42]: image
The arrow is hard to see from the above picture. Is it possible to change the arrow color to different color depends on the dark/light mode?

**Comment by cyr17** [comment]
> Having issues running the client: Screen Shot 2021-10-20 at 9 46 06 pm
> 
> I have ran `npm install` from both the root folder and the client folder

I had this issue once , try deleting node_modules and do npm install again and it might work

**Comment by ghost** [comment]
> > I have ran `npm install` from both the root folder and the client folder
> 
> I had this issue once , try deleting node_modules and do npm install again and it might work

if that doesn't work, `npm install --save draft-js react-draft-wysiwyg` works in my case

**Comment by melvinalex7** [comment]
> It will be great if we can add a darker border to the `<-` and `->` arrows or change the colour of them to darker grey in the layout preview, it is a bit hard to see the arrows in light mode with white background. But overall lgtm

Unfortunately, only React Bootstrap v 2.0.0 allows the change of carousel control colors and the current version ours is v 1.6.1. I'm not quite sure if upgrading the bootstrap will affect other functionalities on main.

**Pull request merged**